tirocks/src/db/imp.rs

```diff
@@ -0,0 +1,726 @@
+                db.get_ptr(),
+                path.as_ref().path_to_slice(),
+                ctx,
+                Some(fp),
+            ))?;
+        }
+        if !convert_s.ok() {
+            return Err(convert_s);
+        }
+        Ok(names)
+    }
+
+    /// Create a column_family.
+    ///
+    /// If it's a titan db, it's the same as calling `create_cf_titan` with the titan specific
+    /// options set to default.
+    pub fn create_cf(&mut self, name: impl AsRef<str>, opt: CfOptions) -> Result<()> {
+        if self.is_titan {
+            return self.create_cf_titan(name, opt.into());
+        }
+        let ptr = unsafe {
+            ffi_call!(crocksdb_create_column_family(
+                self.ptr,
+                opt.as_ptr(),
+                r(name.as_ref().as_bytes()),
+            ))
+        }?;
+        if let Some(c) = opt.comparator() {
+            self.comparator.push(c.clone());
+        }
+        self.handles
+            .push(unsafe { RefCountedCfHandle::from_ptr(ptr, true) });
+        Ok(())
+    }
+
+    /// Create a column_family.
+    ///
+    /// If it's not titan, it's the same as calling `create_cf`. Titan specific options will be
+    /// ignored.
+    pub fn create_cf_titan(
+        &mut self,
+        name: impl AsRef<str>,
+        mut opt: TitanCfOptions,
+    ) -> Result<()> {
+        if !self.is_titan {
+            return self.create_cf(name, opt.into());
+        }
+        let ptr = unsafe {
+            ffi_call!(ctitandb_create_column_family(
+                self.ptr,
+                opt.as_mut_ptr(),
+                r(name.as_ref().as_bytes()),
+            ))
+        }?;
+        if let Some(c) = opt.comparator() {
+            self.comparator.push(c.clone());
+        }
+        self.handles
+            .push(unsafe { RefCountedCfHandle::from_ptr(ptr, true) });
+        Ok(())
+    }
+
+    /// Destroy a column family specified by name. This call records a drop record in the
+    /// manifest and prevents the column family from flushing and compacting.
+    pub fn destroy_cf(&mut self, name: &str) -> Result<bool> {
+        if name == DEFAULT_CF_NAME {
+            return Err(Status::with_invalid_argument(
+                "default cf can't be dropped.",
+            ));
+        }
+        let pos = self
+            .handles
+            .iter()
+            .position(|h| h.name().map_or(false, |n| n == name));
+        let pos = match pos {
+            Some(p) => p,
+            None => return Err(Status::with_code(Code::kNotFound)),
+        };
+        let mut h = self.handles.swap_remove(pos);
+        unsafe {
+            let destroy_res = h.destroy(self.ptr);
+            let drop_res = h.maybe_drop(self.ptr);
+            destroy_res.and(drop_res)
+        }
+    }
+
+    pub fn cfs(&self) -> impl Iterator<Item = &RawCfHandle> {
+        self.handles.iter().map(|c| &**c)
+    }
+
+    pub fn cf(&self, name: &str) -> Option<&RawCfHandle> {
```

Comment:
Is it intended that user can only reference a cf_handle instead of keeping a copy themselves? If that's the case why bother ref counting it?

Reply:
`RawCfHandle` can't be dereferenced. For keeping a reference on their own, `CfHandle::new(db, name)` should be used instead.